addons/pkg/util/clusterbootstrapclone/clusterbootstrapclone.go

```diff
@@ -62,6 +63,10 @@ func NewHelper(ctx context.Context, k8sClient client.Client, aggregateAPIResourc
 // a fully qualified name(<packageShortName>.<domain>.<version>). The fully qualified name is fetched from TKR.
 func (h *Helper) CompleteCBPackageRefNamesFromTKR(tkr *runtanzuv1alpha3.TanzuKubernetesRelease, clusterBootstrap *runtanzuv1alpha3.ClusterBootstrap) error {
 	var suffix = "*"
+
+	if clusterBootstrap.Spec == nil {
+		return errors.New("missing spec definition")
+	}
 	clusterBootstrapPackages := []*runtanzuv1alpha3.ClusterBootstrapPackage{
```

Comment:
hmm. I think Codecov is missing something. This lines do get cover by this test: 
https://github.com/vmware-tanzu/tanzu-framework/pull/4264/files#diff-2d52a425f83e5f6840a1829491d52865109c05aa05a320f36b1eb397c784f53cR487

"Clusterbootstrap is NOT annotated and spec is nil"